src/EditorFeatures/CSharpTest/RemoveUnnecessaryParentheses/RemoveUnnecessaryParenthesesTests.cs

```diff
@@ -1651,5 +1651,26 @@ void M()
     }
 }", offeredWhenRequireForClarityIsEnabled: true);
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsRemoveUnnecessaryParentheses)]
+        public async Task TestIsPatternAndLogical_TestWithAllOptionsSetToIgnore()
+        {
+            await TestAsync(
+@"class C
+{
+    void M(object expression)
+    {
+        if ($$(expression is bool b) && b) { }
+    }
+}",
+@"class C
+{
+    void M(object expression)
+    {
+        if (expression is bool b && b) { }
+    }
```

Comment:
There is already a test showing the parentheses are not removed

Reply:
> ```expression is b && b``` (constant pattern)?

That's shouldn't be a constant pattern.  That's just "foo is string && ..."  If that was a constant pattern, that would break a lot of code.